File: app/src/main/java/com/example/milkaggregatorapplication/MyAdapter.java
package com.example.milkaggregatorapplication;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    ArrayList<User> list;



    public MyAdapter(Context context, ArrayList<User> list) {


        this.list = list;

    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item,parent,false);
        return  new MyViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {

        User user = list.get(position);
        holder.Product.setText(user.getProduct());
        holder.name1.setText(user.getName());
        holder.mobile1.setText(user.getMobile());
        holder.address1.setText(user.getAddress());


        holder.amount.setText(user.getAmount());


    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder{

        TextView Product,amount,name1,mobile1,address1;


        public MyViewHolder(@NonNull View itemView) {
            super(itemView);

            Product = itemView.findViewById(R.id.tvfirstName);

            amount = itemView.findViewById(R.id.tvage);

            name1 = itemView.findViewById(R.id.nameid);
//
//            mobile1 = itemView.findViewById(R.id.mobileid);
//
//            address1 = itemView.findViewById(R.id.addressid);

        }
    }

}

File: app/src/main/java/com/example/milkaggregatorapplication/MyAdapter.kt
package com.example.milkaggregatorapplication

import android.content.Context
import com.example.milkaggregatorapplication.User.product
import com.example.milkaggregatorapplication.User.name
import com.example.milkaggregatorapplication.User.mobile
import com.example.milkaggregatorapplication.User.address
import com.example.milkaggregatorapplication.User.amount
import com.example.milkaggregatorapplication.MyAdapter.MyViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.example.milkaggregatorapplication.R
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList

class MyAdapter(context: Context?, var list: ArrayList<User>) :
    RecyclerView.Adapter<MyViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.item, parent, false)
        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val user = list[position]
        holder.Product.text = user.product
        holder.name1.text = user.name
        holder.mobile1!!.text = user.mobile
        holder.address1!!.text = user.address
        holder.amount.text = user.amount
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var Product: TextView
        var amount: TextView
        var name1: TextView
        var mobile1: TextView? = null
        var address1: TextView? = null

        init {
            Product = itemView.findViewById(R.id.tvfirstName)
            amount = itemView.findViewById(R.id.tvage)
            name1 = itemView.findViewById(R.id.nameid)
            //
//            mobile1 = itemView.findViewById(R.id.mobileid);
//
//            address1 = itemView.findViewById(R.id.addressid);
        }
    }
}